cpg_workflows/inputs.py

```diff
@@ -251,88 +236,83 @@ def _populate_alignment_inputs(
     return None
 
 
-def _populate_analysis(cohort: Cohort) -> None:
+def populate_analysis(dataset: Dataset) -> None:
     """
     Populate Analysis entries.
     """
-    for dataset in cohort.get_datasets():
-        gvcf_by_sgid = get_metamist().get_analyses_by_sgid(
-            dataset.get_sequencing_group_ids(),
-            analysis_type=AnalysisType.GVCF,
-            dataset=dataset.name,
-        )
-        cram_by_sgid = get_metamist().get_analyses_by_sgid(
-            dataset.get_sequencing_group_ids(),
-            analysis_type=AnalysisType.CRAM,
-            dataset=dataset.name,
-        )
+    gvcf_by_sgid = get_metamist().get_analyses_by_sgid(
+        dataset.get_sequencing_group_ids(),
+        analysis_type=AnalysisType.GVCF,
+        dataset=dataset.name,
+    )
+    cram_by_sgid = get_metamist().get_analyses_by_sgid(
+        dataset.get_sequencing_group_ids(),
+        analysis_type=AnalysisType.CRAM,
+        dataset=dataset.name,
+    )
+
+    for sequencing_group in dataset.get_sequencing_groups():
+        if (analysis := gvcf_by_sgid.get(sequencing_group.id)) and analysis.output:
+            # assert file exists
+            assert exists(analysis.output), ('gvcf file does not exist', analysis.output)
+            sequencing_group.gvcf = GvcfPath(path=analysis.output)
+        elif exists(sequencing_group.make_gvcf_path()):
+            logging.warning(
+                f'We found a gvcf file in the expected location {sequencing_group.make_gvcf_path()},'
+                'but it is not logged in metamist. Skipping. You may want to update the metadata and try again. ',
+            )
+        if (analysis := cram_by_sgid.get(sequencing_group.id)) and analysis.output:
+            # assert file exists
+            assert exists(analysis.output), ('cram file does not exist', analysis.output)
+            crai_path = analysis.output.with_suffix('.cram.crai')
+            if not exists(crai_path):
+                crai_path = None
+            sequencing_group.cram = CramPath(analysis.output, crai_path)
+
+        elif exists(sequencing_group.make_cram_path()):
+            logging.warning(
+                f'We found a cram file in the expected location {sequencing_group.make_cram_path()},'
+                'but it is not logged in metamist. Skipping. You may want to update the metadata and try again. ',
+            )
```

Comment:
If I'm not wrong here, if you had multiple files that didn't exist, this code will fail on the first one, and would take you multiple runs to identify all of them. Would it be sensible to change it to make a log of all files that don't exist, and then error at the end if it finds any, and print all the error messages then?

Reply:
Quick disclaimer, this code block is moved but unchanged, so I probably wouldn't alter it in this PR

So yes but also maybe no - The code only hard fails if there's a metamist entry pointing to a non-existent file. We could refactor this to collect all the failures, that would be a fairly simple change. 

The Warning messages are for an unregistered file that DOES exist, and the fallback is leaving these attributes as None, which... is accurate, and it's down to the Stage whether that's important (tolerate or throw an error)

Reply:
Yeah the warnings I am fine with, was just wondering about the failures. But for sure we could leave this bit of refactoring to after the migration.

Reply:
This block actually needs a refactor anyway, given that since the first-class files change ([here](https://github.com/populationgenomics/metamist/pull/856)) the analysis entries in Metamist do still have `X.output`, but we also have a more full-featured `X.outputs.path` and a bunch of derivative attributes like file size and integrity hashes. We should probably upgrade to use that.